src/taglibs/migrate/util/addIdScopedAttr.js

```diff
@@ -0,0 +1,16 @@
+module.exports = function addIdScopedAttr(context, el, attrValue) {
+    const templateRoot = el.parentNode;
+    const walker = context.createWalker({
+        enter(node) {
+            if (node.type == "HtmlElement" && node.hasAttribute) {
```

Comment:
Looks good but I think that we also need to check for `type === "CustomTag"`. It would be simplest to only check for `node.hasAttribute` as only `HTMLElement`'s and `CustomTags`'s should have that method.